internal/player/module.go

```diff
@@ -12,20 +12,23 @@ var Module = fx.Provide(
 	NewHlsPlaylistHandler,
 	NewHlsFragmentHandler,
 	NewHlsPlayerHandler,
+	NewHlsManifestHandler,
 )
 
 func Run(
 	lifecycle fx.Lifecycle,
 	playlistHandler *HlsPlaylistHandler,
 	fragmentHandler *HlsFragmentHandler,
 	playerHandler *HlsPlayerHandler,
+	manifestHandler *HlsManifestHandler,
 ) {
 	lifecycle.Append(fx.Hook{
 		OnStart: func(ctx context.Context) error {
 			r := mux.NewRouter()
 			r.Handle("/playlist/{filename}", playlistHandler).Methods("GET")
 			r.Handle("/fragment/{filename}", fragmentHandler).Methods("GET")
 			r.Handle("/player", playerHandler).Methods("GET")
+			r.Handle("/manifest", manifestHandler).Methods("GET")
```

Comment:
I did not understand the reason behind this endpoint, sorry. Could you explain?

Reply:
Also, I think we can use just one handler and pass the request path to the storage. What is your opinion?

Reply:
I created this new endpoint to serve a custom json with all mosaics playlist.m3u8 and can possible to create a button for each mosaic registered:

Manifest response based in tasks.json:
> GET http://localhost:8090/manifest
```json
[
    {
        "name": "mosaic",
        "playlist_url": "http://localhost:8090/playlist/playlist-mosaic.m3u8"
    },
    {
        "name": "mosaic2",
        "playlist_url": "http://localhost:8090/playlist/playlist-mosaic2.m3u8"
    }
]
```

On HLS player, i can mount this:

![Buttons to select a mosaic](https://github.com/mauricioabreu/mosaic-video/assets/34068867/72811a7e-7a5d-40f3-a49f-c49c59352c86)

When user click on buttom, the HLS player make a request to `http://localhost:8090/playlist/playlist-mosaic.m3u8`, load manifest with all fragments and automatically make a request for each fragment into endpoint: `http://localhost:8090/fragment/fragment-name.ts`, therefore i need theses 3 endpoints:

- r.Handle("/manifest", manifestHandler): to serve a list of mosaics with name and playlist url;
- r.Handle("/playlist/{filename}", playlistHandler): to serve a specific playlist.m3u8 containing the fragments;
- r.Handle("/fragment/{filename}", fragmentHandler): to serve a specific fragment.ts.

Reply:
But, now i revised and HLS player use same endpoint to get playlist and fragment (/player/playlist/filename), then we need only two endpoints:
- GET /playlist 
- GET /manifest